datastream-utils/src/main/java/com/linkedin/datastream/common/ThreadFuture.java

```diff
@@ -0,0 +1,105 @@
+/**
+ *  Copyright 2019 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.common;
+
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang3.Validate;
+
+
+/**
+ * A {@link Future} impl that wraps a {@link Thread}.
```

Comment:
Can you please document the benefit of converting thread to future instead of directly checking the thread status?

Reply:
+1

Reply:
There's no inherent benefit. We're just adapting one API to another, i.e. we have a utility (`FutureUtils`) that checks if a `Future` is done after a `timeout` and we're using that to verify if connector tasks stop within a specific duration. We need that same logic for situations when we working with threads not futures.

My initial intent was to keep `ThreadFuture` private within the MySQL connector module but you suggested it may be useful as a general purpose utility.